django/template/engine.py

```diff
@@ -27,12 +28,15 @@ def __init__(self, dirs=None, app_dirs=False, context_processors=None,
             loaders = ['django.template.loaders.filesystem.Loader']
             if app_dirs:
                 loaders += ['django.template.loaders.app_directories.Loader']
+            if post_app_dirs:
+                loaders += [('django.template.loaders.filesystem.Loader', post_app_dirs)]
             if not debug:
                 loaders = [('django.template.loaders.cached.Loader', loaders)]
         else:
             if app_dirs:
-                raise ImproperlyConfigured(
-                    "app_dirs must not be set when loaders is defined.")
+                raise ImproperlyConfigured('app_dirs must not be set when loaders is defined.')
+            if post_app_dirs:
```

Comment:
If you have a filesystem loader it will (likely) be used. I guess it's unlikely someone would set DIRS and override loaders but forget a fliesystem loader.